### PR TITLE
Fix social media validation for Twitter

### PR DESCRIPTION
**Pull request by abbeycode**
Using change from CocoaPods/Core#571 to address Issue CocoaPods/CocoaPods#9049

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Fix social media validation for Twitter  
  [abbeycode](https://github.com/abbeycode)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lib/cocoapods/validator.rb [hunk 1]
```diff
@@ -450,7 +450,7 @@ def validate_screenshots(spec)
     # Performs validations related to the `social_media_url` attribute.
     #
     def validate_social_media_url(spec)
-      validate_url(spec.social_media_url) if spec.social_media_url
+      validate_url(spec.social_media_url, 'Not a Browser') if spec.social_media_url
```
Comment (dnkoutso) [inline lib/cocoapods/validator.rb]: should this be 'CocoaPods' or is this standard practice for unknown/placeholder user agents?

File: lib/cocoapods/validator.rb [hunk 2]
```diff
@@ -416,7 +416,7 @@ def perform_extensive_subspec_analysis(spec)
 
     # Performs validation of a URL
     #
-    def validate_url(url)
+    def validate_url(url, user_agent=nil)
```
Comment (dnkoutso) [inline lib/cocoapods/validator.rb]: nit formatting: `user_agent = nil`

I think Rubocop will catch that anyway.

**Comment by abbeycode** [comment]
I'll take a look at the broken test specs tomorrow.

**Comment by abbeycode** [comment]
@dnkoutso This is all set now, if you'd like to review again.

**Comment by dnkoutso** [comment]
@abbeycode the Core PR has been merged please update the `Gemfile.lock` to point to the right SHA and not your fork and we can merge this!

**Comment by dnkoutso** [comment]
Also please squash into a single commit.

**Comment by abbeycode** [comment]
@dnkoutso I'm curious what the reason for squashing is, since GitHub lets you squash when you merge. I'm happy to do it, but I was curious if there's a reason that applies to the projects I maintain.

**Comment by dnkoutso** [comment]
No big reason its just that I forget to update the merge mode :) sorry

**Comment by abbeycode** [comment]
All set, once the checks complete.

**Comment by dnkoutso** [comment]
Looks like there was no squash anyway i will make sure its squashes into a single commit once merging.

**Pull request merged**